simpleranker.py

```diff
@@ -50,15 +50,21 @@ def rank(self, docs: DocumentArray, parameters: Dict, *args, **kwargs):
 
         for doc in docs[traversal_paths]:
             matches_of_chunks = []
-            matches_of_chunks.extend(doc.matches)
+            matches_no_parent = []
+            for m in doc.matches:
+                if m.parent_id is not None:
+                    matches_of_chunks.append(m)
+                else:
+                    matches_no_parent.append(m)
             doc.matches.clear()
             for chunk in doc.chunks:
                 matches_of_chunks.extend(chunk.matches)
             groups = groupby(
                 sorted(matches_of_chunks, key=lambda d: d.parent_id),
                 lambda d: d.parent_id,
             )
```

Comment:
what about

```
    groups = groupby(
                sorted(matches_of_chunks, key=lambda d: d.parent_id or d.id),
                lambda d: d.parent_id or d.id,
            )
```

Reply:
Yes this was my first thought but decided to take a more conservative way. Will double-test

Reply:
you are right this one actually works

Reply:
failing CI, needs more work